packages/eslint-plugin-ecmascript-compat/lib/features/es2021.js

```diff
@@ -12,6 +12,7 @@ module.exports = [
       options: noRestrictedSyntaxPrototypeMethod('String.prototype.replaceAll', 'ES2021'),
     },
     compatFeatures: [compatData.javascript.builtins.String.replaceAll],
+    polyfill: 'String.prototype.replaceAll',
   },
```

Comment:
`Promise.any` below should also be polyfillable.

I checked against this [list](https://github.com/robatwilliams/es-compat/blob/master/packages/eslint-plugin-ecmascript-compat/lib/features/es-versions.md) and you've got all the others.